lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -1539,28 +1555,26 @@ struct HandshakeInsertBufferPass
               return !isa<handshake::BufferOp>(operand.getOwner());
             }));
       }
-      // For unvisited operations, recursively call insertBufferOp() method.
-      else if (!opVisited[user])
-        insertBufferOp(user, builder);
+      // For unvisited operations, recursively call insertBufferDFS() method.
+      else if (opVisited.count(user) == 0)
+        insertBufferDFS(user, builder, opVisited, opOnStack);
     }
     // Pop operation out of the stack.
-    opOnStack[op] = false;
+    opOnStack.erase(op);
   }
 
   void runOnOperation() override {
-    auto f = getOperation();
-    for (auto &block : f) {
-      for (auto &op : block) {
-        opVisited[&op] = false;
-        opOnStack[&op] = false;
-      }
-    }
-    // Traverse each use of each argument of the entry block.
-    auto builder = OpBuilder(f.getContext());
-    for (auto &arg : f.getBody().front().getArguments()) {
-      for (auto &operand : arg.getUses()) {
-        if (!opVisited[operand.getOwner()])
-          insertBufferOp(operand.getOwner(), builder);
+    if (strategies.size() == 0)
+      strategies = {"cycles"};
```

Comment:
Why not have a default on the option, rather than checking for an empty list here?

Reply:
It looks like default options don't work for `cl::list`. I think keeping this is preferable to switching to a scalar option with a default and parsing it here.